src/modules/module.cpp

```diff
@@ -122,6 +125,36 @@ void MtbModule::jsonSetConfig(QTcpSocket*, const QJsonObject &json) {
 		this->name = json["name"].toString();
 }
 
+void MtbModule::jsonSetAddress(QTcpSocket *socket, const QJsonObject &request) {
+	if (this->isFirmwareUpgrading()) {
+		sendError(socket, request, MTB_MODULE_UPGRADING_FW, "Firmware of module is being upgraded!");
+		return;
+	}
+	if (this->busModuleInfo.inBootloader()) {
+		sendError(socket, request, MTB_MODULE_IN_BOOTLOADER, "Module is in bootloader!");
+		return;
+	}
+	if (this->isConfigSetting()) {
+		sendError(socket, request, MTB_MODULE_CONFIG_SETTING, "Configuration of module is being changed!");
+		return;
+	}
+	uint8_t newaddr;
+	newaddr = request["new_address"].toInt(1);
+	if (request.contains("address")) {
+		mtbusb.send(
+			Mtb::CmdMtbModuleChangeAddr(
+				this->address, newaddr
+			)
+		);
+	} else {
+		mtbusb.send(
+			Mtb::CmdMtbModuleChangeAddr(
+				newaddr
+			)
+		);
+	}
```

Comment:
Klidně bych tady kód zkomprimoval na méně řádků, ale jak chceš.

Reply:
No snažil jsem se držet tvoje šílené konvence, kterým teda vůbec nerozumím. Resp. nevím co komu vlastně přináší.